eventsourcing/src/main/java/org/axonframework/eventsourcing/annotation/EventSourcedEntity.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventsourcing.annotation;
+
+import org.axonframework.eventsourcing.CriteriaResolver;
+import org.axonframework.eventsourcing.EventSourcingRepository;
+import org.axonframework.eventsourcing.eventstore.EventCriteria;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation to configure several aspects of an event-sourced entity. This annotation is required to construct a
+ * {@link AnnotationBasedEventSourcingEntityRepository}.
+ * <p>
+ * While loading the entity from the {@link EventSourcingRepository}, the provided {@code id} needs to be translated to
+ * an {@link EventCriteria} instance to load the correct events. Unless overridden, this translation is done by the
+ * {@link AnnotationBasedEventCriteriaResolver}. So, {@link EventCriteria} will be resolved in the following order:
+ * <ol>
+ *     <li>
+ *         Using the custom {@link #criteriaResolver()}. The provided class should implement the {@link CriteriaResolver},
+ *         and have a no-arg constructor, or a 1-arg constructor with the {@link Class} of the entity as parameter.
+ *     </li>
+ *     <li>
+ *         By defining a static method in the entity class annotated with {@link EventCriteriaBuilder} which returns an
+ *         {@link EventCriteria} and accepts the {@code id} as a parameter. This method should be static and return an
+ *         {@link EventCriteria}. Multiple methods can be defined with different id types, and the first matching method
+ *         will be used.
+ *     </li>
+ *     <li>
+ *         If no matching {@link EventCriteriaBuilder} is found, the {@link EventSourcedEntity#tagKey()} will be used as the tag key, and the {@link Object#toString()} of the id will be used as value.
+ *     </li>
+ *     <li>
+ *         If the {@link EventSourcedEntity#tagKey()} is empty, the {@link Class#getSimpleName()} of the entity will be used as tag key, and the {@link Object#toString()} of the id will be used as value.
```

Comment:
We still haven't decided on the Tag format, so I'm still not sure if it could be class name or (for example) "lower cased class name".

Reply:
Wouldn't hurt to refer to issue [#3326](https://github.com/AxonFramework/AxonFramework/issues/3326) here indeed. Once we have the format, we'd need to change documentation in several places already. Such a number reference will make that easier.